src/Integration.Vsix/Integration.Vsix.csproj

```diff
@@ -70,7 +70,6 @@
       <DesignTime>True</DesignTime>
       <DependentUpon>SqmOleCommandTarget.tt</DependentUpon>
     </Compile>
-    <Compile Include="Telemetry\SonarLintIntegrationPackage.IOleCommandTarget.cs" />
```

Comment:
The question I was trying to understand was "how does VS know where to route the command for handling?".

The vsct is registered against this package so VS will route the commands to the package. Before, the package passed them on to _SonarLintSqmCommandTarget_ so the commands were handled.

Now, the package no longer handles the pseudo-commands. How does does VS know to route the commands to _TelemetryLogger_? 
The answer appears to be that it doesn't: on debugging through the new version, the IOleCommandTarget method implementations that were moved to _TelemetryLogger_ are never called.

"_There's an assert_": the SQM facade to raise the events could be called from the Team Explorer before the package is loaded, so the SQM facade for _calling_ the dummy commands does need to be initialised using MEF (I'm guessing that this is the assert you saw).

However, the code to _handle_ the dummy commands doesn't need to be initialised along with the SQM facade when a command is first executed in the Team Explorer - if the commands are being handled by the package, then VS will loaded the package at that point and ask it handle the command.

The simplest fix is to separate the pseudo-command raising from the pseudo-command handling
i.e. revert the command handling so the commands really are handled by the package.
The SQM facade does need to be initialised via MEF so that it can be used the Team Explorer. This also works ok for the package when it (indirectly) raises the _BoundSolutionDetected_ pseudo event.